lib/Transforms/Scalar/UnifyAssumes.cc

```diff
@@ -48,48 +56,68 @@ class UnifyAssumesPass : public ModulePass {
 
 char UnifyAssumesPass::ID = 0;
 
-bool isAssumeCall(const CallInst &ci, seahorn::SeaBuiltinsInfo &SBI) {
+bool UnifyAssumesPass::isAssumeCall(const CallInst &ci) {
   using namespace seahorn;
-  switch(SBI.getSeaBuiltinOp(ci)) {
+  switch (m_SBI->getSeaBuiltinOp(ci)) {
   default: return false;
   case SeaBuiltinsOp::ASSUME:
   case SeaBuiltinsOp::ASSUME_NOT:
     return true;
   }
 }
+
+bool UnifyAssumesPass::isAssertCall(const CallInst &ci) {
+  using namespace seahorn;
+  switch (m_SBI->getSeaBuiltinOp(ci)) {
+  default:
+    return false;
+  case SeaBuiltinsOp::ASSERT:
+  case SeaBuiltinsOp::ASSERT_NOT:
+    return true;
+  }
+}
+
 /// Find a function exit basic block.  Assumes that the function has
 /// a unique block with return instruction
-llvm::BasicBlock *findExitBlock(llvm::Function &F) {
+llvm::BasicBlock *UnifyAssumesPass::findExitBlock(llvm::Function &F) {
```

Comment:
why is this function a method? used to be just a local function.

Reply:
In the previous review you asked me to change SBI to a class member and use it in one of the methods. I assumed that you wanted to pass this to all functions that use SBI and making everything a method was the natural conclusion i drew. Also the mix of local functions and methods seemed arbitrary AFAICT. Is there a reason why this can't be a function?

Reply:
`findExitBlock` does not use `SBI`. If it does, it should be a method.

Better use the simplest concept that works. If a function is just a function, it should not be part of a class. This is not Java. Often class will have functions that only the class uses, these can be hidden inside cpp files. Sometimes a function, like `findExitBlock` is waiting to be promoted to be used by other clients as well.

Reply:
reverted change to `findExitBlock`